packages/ui/src/services/icon-config/types.ts

```diff
@@ -1,110 +1,40 @@
-   * {
-   *  name: /md (.*)/,
-   *  content: (icon: string) => icon,
-   *  iconClass: 'material-icons'
-   *  color: '#eee' // Additional styling for icon
-   * }
-   * ```
-   * `<va-icon name="md alarm"/>` ->
-   *
-   * `<i class="md" style="color: #eee;">alarm</i>`
-   *
-   * @see https://material.io/resources/icons/?icon=alarm&style=baseline
-   */
-  content?: string | ((...regexGroups: string[]) => string | undefined)
+export interface IconConfigurationDefaultParams {
+  attrs?: Record<string, string | ((...args: any[]) => unknown)>
 
   /** Vue component */
-  component?: any,
-
-  /**
-   * You can use custom component, bind some props to it.
-   * @requires component
-   *
-   * @example
-   * ```
-   * {
-   *  name: /(brandico, eva, ant-design) (.*)/,
-   *  component: IconifyComponent,
-   *  componentProps: (font: string, icon: string) => ({ icon: `${font}:${icon}` })
-   *  color: '#eee' // Additional styling for icon
-   * }
-   * ```
-   *
-   * IconifyComponent example component:
-   * ```html
-   * <template>
-   *  <div class="iconify" :data-icon="icon" data-inline="false" />
-   * </template>
-   *
-   * <script>
-   * export defualt {
-   *  props: ['icon']
-   * }
-   * </script>
-   * ```
-   *
-   * @see https://iconify.design/icon-sets/brandico/
-   */
-  componentProps?: Record<string, any> | ((...regexGroups: string[]) => Record<string, any>)
-
-  /**
-   * Use this to create icon that u will often to use.
-   *
-   * Example:
-   * ```js
-    {
-      name: 'twitter',
-      to: 'brandico twitter-bird',
-      color: '#1da1f2',
-    },
-    { // Or for Yandex
-      name: 'yandex',
-      to: 'brandico yandex',
-      color: '#ff0000',
-    },
-    {
-      name: 'vuestic-logo',
-      to: VuesticLogoComponent,
-      color: '#62e471',
-    }
-  * ```
-  */
-  to?: string
-
-  // PROPS
+  component?: any
+  /** Content that will be passed as text inside icon tag or as slot in component */
+  content?: string
 
+  class?: string,
   tag?: string
   color?: string
   rotation?: number | string
   spin?: 'clockwise' | 'counter-clockwise'
+  to?: string
+}
+
+export interface IconConfigurationWithDynamicSegmentName extends IconConfigurationDefaultParams {
+  name: IconNameWithDynamicSegment
+  resolve?: ((dynamicSegmentsValue: {[dynamicSegmentName: string]: string }) => IconConfigurationDefaultParams)
 }
 
-export type IconsConfig = IconConfig[]
+export interface IconConfigurationWithRegexName extends IconConfigurationDefaultParams {
+  name: RegExp
+  // Need a different resolve method name because ts don't understand types
+  resolveFromRegex?: ((...regexGroupValues: string[]) => IconConfigurationDefaultParams)
+}
+
+export type IconConfiguration = IconConfigurationWithDynamicSegmentName | IconConfigurationWithRegexName
+
+export type IconConfig = IconConfiguration[]
+
+export function isIconConfigurationWithDynamicSegmentName (config: IconConfiguration): config is IconConfigurationWithDynamicSegmentName {
```

Comment:
I would recommend to just perform a check.

So 

```ts
const configType = getConfigType(config) // 'regex' | 'string', throw if invalid
if (configType === 'regex') {

}
if (configType === 'string') {

}
```

will be vastly more readable